invenio_files_rest/__init__.py

```diff
@@ -6,7 +6,819 @@
+
+   DELETE /files/<bucket_id>/<file_name>
+
+Hard deletion
+^^^^^^^^^^^^^
+Given a specific object version, it will delete the ObjectVersion,
+the referenced FileInstance and the file on disk. If the deleted version
+was the :code:`head`, it will then set the previous object
+as the new head.
+
+The deletion of files on disk will not happen immediately. This is because
+it is done via an asynchronous task to ensure that the FileInstance is
+safely removed from the database in case the low level operation of file
+removal on disk fails for any unexpected reason.
+
+You can hard delete a file using REST APIs:
+
+.. code-block:: console
+
+   DELETE /files/<bucket_id>/<file_name>?versionId=<version_id>
+
+REST APIs do not allow to perform delete operations that can affect multiple
+objects at the same time. For advanced use cases, you will to use the
+Invenio-Files-REST APIs programmatically.
+
+Authorization
+-------------
+Invenio-Files-REST relies on `Invenio-Access
+<https://invenio-access.readthedocs.io>`_ to implement files authorization.
+The following documentation assumes that you already have knowledge of how
+authorization works on Invenio.
+
+Invenio-Files-REST defines a set of actions for operations on Bucket and
+ObjectVersions that can be used to implement authorization as you need:
+
+    - files-rest-location-update
+    - files-rest-bucket-read
+    - files-rest-bucket-read-versions
+    - files-rest-bucket-update
+    - files-rest-bucket-listmultiparts
+    - files-rest-object-read
+    - files-rest-object-read-version
+    - files-rest-object-delete
+    - files-rest-object-delete-version
+    - files-rest-multipart-read
+    - files-rest-multipart-delete
+
+Response codes
+^^^^^^^^^^^^^^
+If the authorization for an action fails, Invenio-Files-REST normally returns
+a :code:`403` response code for authenticated users, :code:`401` otherwise.
+For security reasons, when trying to retrieve an unauthorized file, it will
+return a :code:`404` instead to hide the existence or
+non-existence of the file.
+
+Authorization definition
+^^^^^^^^^^^^^^^^^^^^^^^^
+The default permission factory
+:py:data:`invenio_files_rest.permissions.permission_factory` will authorize
+users that has :code:`Needs` that fulfill the actions listed above. This means
+that by default no user will be authorized (with the exception of
+any :code:`superuser`).
+
+Depending on how you are planning to integrate Invenio-Files-REST in your
+Invenio application, you might want to decide how to give permissions for
+operations on files.
+
+If you plan to give authorization to specific users or roles, you can use the
+default permission factory and assign user or roles to the actions listed
+above as described in the Invenio-Access documentation.
+
+If instead you want to define permissions based on other object, for example
+on records to which the files are attached to, then you will have to
+define your own permission factory and used via the configuration variable
```

Comment:
```suggestion
define your own permission factory and use it via the configuration variable
```
Not pretty sure about this one.